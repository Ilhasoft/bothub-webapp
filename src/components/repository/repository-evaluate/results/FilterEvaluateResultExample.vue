<template>
  <div class="filter-evaluate-example bh-grid">
    <bh-field
      :errors="errors.intent"
      class="bh-grid__item bh-grid__item--grow-3">
      <bh-text
        v-model="text"
        :debounce="debounceTime">
        <div slot="append">
          <bh-icon-button
            value="magnify"
            size="small"
            type="submit" />
        </div>
      </bh-text>
    </bh-field>
    <div class="bh-grid__item filter-evaluate-example__text">
      <span>Filter by:</span>
    </div>
    <bh-field
      :errors="errors.intent"
      class="bh-grid__item">
      <bh-autocomplete
        v-model="intent"
        :data="intents || []"
        :formatters="inputFormatters"
        placeholder="All Intent" />
    </bh-field>
    <bh-field
      :errors="errors.label"
      class="bh-grid__item">
      <bh-autocomplete
        v-model="label"
        :data="labels || []"
        :formatters="inputFormatters"
        placeholder="All labels" />
    </bh-field>
    <bh-field
      :errors="errors.intent"
      class="bh-grid__item">
      <bh-autocomplete
        v-model="entity"
        :data="entities || []"
        :formatters="inputFormatters"
        placeholder="All entities" />
    </bh-field>
  </div>
</template>

<script>
import { formatters } from '@/utils/index';
import _ from 'lodash';


export default {
  name: 'FilterEvaluateResultExample',
  props: {
    debounceTime: {
      type: Number,
      default: 750,
    },
    intents: {
      type: Array,
      default: null,
    },
    entities: {
      type: Array,
      default: null,
    },
    labels: {
      type: Array,
      default: null,
    },
  },
  data() {
    return {
      text: '',
      intent: '',
      label: '',
      entity: '',
      setTimeoutId: null,
      errors: {},
    };
  },
  computed: {
    inputFormatters() {
      const formattersList = [
        formatters.bothubItemKey(),
      ];
      formattersList.toString = () => 'inputFormatters';
      return formattersList;
    },
  },
  watch: {
    text(value) {
      this.$emit('queryStringFormated', { search: value });
    },
    intent: _.debounce(function emitIntent(value) {
      this.$emit('queryStringFormated', { intent: value });
    }, 500),
    entity: _.debounce(function emitEntity(value) {
      this.$emit('queryStringFormated', { entity: value });
    }, 500),
  },
};
</script>

<style lang="scss">
.filter-evaluate-example {
  align-items: center;
  justify-content: center;
  text-align: right;

  &__text {
    margin-top: .5rem;
  }
}
</style>
