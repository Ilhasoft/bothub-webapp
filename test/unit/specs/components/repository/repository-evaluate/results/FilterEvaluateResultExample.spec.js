import { shallowMount, createLocalVue } from '@vue/test-utils';
import FilterEvaluateResultExample from '@/components/repository/repository-evaluate/results/FilterEvaluateResultExample';
import BH from 'bh';


const localVue = createLocalVue();
localVue.use(BH);

describe('FilterEvaluateResultExample.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(FilterEvaluateResultExample, {
      localVue,
      propsData: {
        debounceTime: 2,
        intents: [],
        entities: [],
        labels: [],
      },
    });
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
