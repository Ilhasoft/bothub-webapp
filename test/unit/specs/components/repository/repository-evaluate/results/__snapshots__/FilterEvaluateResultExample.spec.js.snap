// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterEvaluateResultExample.vue renders 1`] = `
<div class="filter-evaluate-example bh-grid">
  <bh-field-stub errors="" class="bh-grid__item bh-grid__item--grow-3">
    <bh-text-stub debounce="2" value="" formatters="">
      <div>
        <bh-icon-button-stub type="submit" value="magnify" size="small"></bh-icon-button-stub>
      </div>
    </bh-text-stub>
  </bh-field-stub>
  <div class="bh-grid__item filter-evaluate-example__text"><span>Filter by:</span></div>
  <bh-field-stub errors="" class="bh-grid__item">
    <bh-autocomplete-stub debounce="0" value="" formatters="inputFormatters" data="" placeholder="All Intents"></bh-autocomplete-stub>
  </bh-field-stub>
  <bh-field-stub errors="" class="bh-grid__item">
    <bh-autocomplete-stub debounce="0" value="" formatters="inputFormatters" data="" placeholder="All labels"></bh-autocomplete-stub>
  </bh-field-stub>
  <bh-field-stub errors="" class="bh-grid__item">
    <bh-autocomplete-stub debounce="0" value="" formatters="inputFormatters" data="" placeholder="All entities"></bh-autocomplete-stub>
  </bh-field-stub>
</div>
`;
